src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,58 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+
+    public Figure getRandomFigure() {
+        Figure randomFigure;
+        int randomIndex = new Random().nextInt(5);
+        switch (randomIndex) {
+            case 0 -> randomFigure = getRandomCircle();
+            case 1 -> randomFigure = getIsoscelesTrapezoid();
+            case 2 -> randomFigure = getRandomRectangle();
+            case 3 -> randomFigure = getRightTriangle();
+            case 4 -> randomFigure = getRandomSquare();
+            default -> randomFigure = new Figure();
+        }
+        return randomFigure;
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(Color.WHITE.name(), FigureName.CIRCLE.name(), 10);
+    }
+
+    private Circle getRandomCircle() {
+        return new Circle(new ColorSupplier().getRandomColor(),
+                FigureName.CIRCLE.name(),
```

Comment:
Consider using the `Color` and `FigureName` enums directly instead of calling `name()`. This will maintain type safety and consistency with the enum usage.